include/zephyr/drivers/pcie/pcie.h

```diff
@@ -36,6 +37,23 @@ typedef uint32_t pcie_bdf_t;
  */
 typedef uint32_t pcie_id_t;
 
+struct pcie_dev {
+	pcie_bdf_t bdf;
+	pcie_id_t  id;
+};
+
+#define Z_DEVICE_PCIE_NAME(node_id) _CONCAT(pcie_dev_, DT_DEP_ORD(node_id))
+
+#define PCIE_DT_ID(node_id) PCIE_ID(DT_PROP_OR(node_id, vendor_id, 0xffff), \
+				    DT_PROP_OR(node_id, device_id, 0xffff))
+
+#define DEVICE_PCIE_DECLARE(node_id)                                        \
+	STRUCT_SECTION_ITERABLE(pcie_dev, Z_DEVICE_PCIE_NAME(node_id)) = {  \
+		.id = PCIE_DT_ID(node_id),                                  \
```

Comment:
We need to set also `.bdf = PCIE_BDF_NONE`

Reply:
Thanks. Fixed.